Feat/first challenge

alain-chevanier opened the pull request:
First Challenge solution.

<img width="723" alt="screen shot 2018-07-09 at 2 07 53 am" src="https://user-images.githubusercontent.com/16228995/42435731-0fe46512-831d-11e8-88fa-41633f300bb5.png">
<img width="720" alt="screen shot 2018-07-09 at 2 08 31 am" src="https://user-images.githubusercontent.com/16228995/42435732-0ffd243a-831d-11e8-994d-ae1a673de8f4.png">

Commit:
BASE COMPONENTS AND ROUTING ENABLED

Commit:
REDUX STATE DEFINITION ADDED

Commit:
ACTIVATING AIRBNB ESLINT

Commit:
INTEGRATING REDUX STATE TO ALL COMPONENTS

Commit:
FETCH DATA BEFORE THE COMPONENT IS RENDERED

bgilm commented on src/AppRouting.jsx:
```diff
@@ -0,0 +1,16 @@
+
+const AppRouting = () => (
+  <Router>
+    <Switch>
```
👍

bgilm commented on src/AppRouting.jsx:
```diff
@@ -0,0 +1,16 @@
+    <Switch>
+      <Route path="/trending" component={Trending} />
+      <Route path="/favorite" component={Favorite} />
+      <Redirect to="/trending" />
```
It would be nice to send an object to let the user know what happened with that not found route.

Commit:
SEARCH IN TRENDING GIFS NOW SEARCH FOR RESULTS IN THE SEARCH API OF GIPHY
- NEW SAGA DEFINED TO FETCH DATA ON SEARCH
- NEW ACTION DEFINED TO FETCH DATA FROM API ON SEARCH
- TRENDING COMPONENT SEPARATED TO HANDLE CONCERN OF REDUX BINDING IN A SEPARETE COMPONENT

Commit:
NOW GIF CONTAINER COMPONENT AND GIF COMPONENT MARK THE FAVORITE GIFS

Commit:
FAVORITE CONTAINER NOW FILTERS GIFS LOCALLY AND THE COMPONENT ITSELF NOW WAS SEPARATE ACCORDING TO CONCERNS

Commit:
NOW THE FAVORITE GIFS ARE PERSISTED IN THE LOCAL STORAGE

Commit:
ADDING STYLED NAVIGATION BAR

Commit:
BUGFIX ON THE TRENDING GIFS REDUCERS

Commit:
USING NAVBAR IN FAVORITE AND TRENDING COMPONENTS

Commit:
STYLING THE GIF COMPONENT

Commit:
TEST DEFINED FOR REDUCERS

Commit:
SNAPSHOT TEST ADDED FOR GIF COMPONENT

Commit:
BUGFIX ON UNIT TEST FOR TRENDING GIF REDUCER

Commit:
BUG FIXES ON SNAPSHOT TESTS FOR THE GIF COMPONENT
- ALSO ADDING TESTS FOR ON CLICK EVENT HANDLERS

Commit:
ADDING TEST FOR GIF CONTAINER

timgivois commented on src/components/NavBar/NavBar.style.js:
```diff
@@ -0,0 +1,31 @@
+const NavBarElement = styled(Link)`
+  float: left;
+  color: ${props => (props.active ? 'white' : '#f2f2f2')};
+  background-color: ${props => (props.active ? '#4CAF50' : 'inherit')};
```
👍

timgivois commented on src/index.js:
```diff
@@ -1,8 +1,13 @@
+
 import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
```
still using global css

timgivois commented on src/components/Gif/__snapshots__/Gif.test.jsx.snap:
```diff
@@ -0,0 +1,89 @@
+  text-align: center;
+}
+
+.c2 {
```
:uff:

Commit:
TESTS ADDED FOR APP ROUTING COMPONENT

Commit:
TEST ADDED FOR GIF GRID COMPONENT

Commit:
TEST ADDED FOR ACTION CREATORS
- FAVORITE GIFS
- FILTER TEXT
- TRENDING GIFS

Commit:
TESTS ADDED FOR COMPONENT NAVBAR

Commit:
MOCKING GIF COMPONENT IN GIF GRID TEST SUITE

Commit:
TESTS ADDED FOR TEMPLATE COMPONENT

Commit:
TEST ADDED TO TRENDING COMPONENT

Commit:
REMOVING COVERAGE FOR CONFIGURATION FILES.
- INTEGRATION FILES LIKE INDEX.JS
- STORE DEFINITION
- SAGAS

Commit:
TEST ADDED FOR FAVORITE CONTAINER

Commit:
TEST ADDED FOR FAVORITE GIFS COMPONENT
